samza-core/src/main/java/org/apache/samza/storage/TaskSideInputStorageManager.java

```diff
@@ -242,7 +255,9 @@ private void writeOffsetFiles() {
               .collect(Collectors.toMap(Function.identity(), lastProcessedOffsets::get));
 
             try {
-              String fileContents = OBJECT_MAPPER.writeValueAsString(offsets);
+              String fileContents = OBJECT_MAPPER.writerWithType(
```

Comment:
1. Is the writer required if we use SamzaObjectMapper? What happens if we use SamzaObjectMapper directly?
2. Can we save the ObjectWriter as a static field?

Reply:
1. It uses the default map key and value serializers since its unable to get the type of the key and value. During the deserialization part, it uses the type specific deserializer which expects the string format to be different from the `toString` representation of `SystemStreamPartition` and fails to load it correctly.
2. Done